components/camel-coap/src/main/java/org/apache/camel/coap/CamelCoapResource.java

```diff
@@ -141,10 +146,14 @@ public void handleRequest(Exchange exchange) {
         } catch (Exception e) {
             cexchange.respond(ResponseCode.INTERNAL_SERVER_ERROR, e.getMessage());
         } finally {
-            if (camelExchange != null) {
-                consumer.doneUoW(camelExchange);
+            if (consumer != null) {
+                if (camelExchange != null) {
+                    consumer.doneUoW(camelExchange);
+                }
+                consumer.releaseExchange(camelExchange, false);
+            } else {
+                LOG.error("null in consumer");
```

Comment:
What can be added to this message?

Reply:
Same comment as above: what does `"null in consumer"` tells the user about the conditions or causes that lead to the situation? So, I think that describing the failure as well as the condition provides a better understanding for the user. 

Therefore, something like: `LOG.warn("Skipping releasing the consumer exchange because the consumer is null. It may have been closed/released/etc earlier while trying to do something");`

Reply:
To be clear `It may have been closed/released/etc earlier while trying to do something` should be replaced to the actual conditions that could have nullified it previously.